pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -268,6 +238,29 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		return nil
 	}
 
+	// If there are some invalid feature gates of registration, will output condition
+	// `ValidRegistrationFeatureGates` False in Klusterlet.
+	featureGates, condition, err := helpers.CheckRegistrationFeatureGates(helpers.OperatorTypeKlusterlet,
```

Comment:
With that, there are 5 return values, which is a bit too many for me.

Reply:
if we remove the `error` return value, and merge these two, it would be
```
CheckFeatureGates(operatorType string, registrationConfiguration *operatorapiv1.RegistrationConfiguration,
	workConfiguration *operatorapiv1.WorkConfiguration) (registrationFeatureGates []string, 
	workFeatureGates []string, registrationCondition metav1.Condition, workCondition metav1.Condition)
```
I feel it's a little bit more complicated

Reply:
I think we should have only one condition.